### DISCUSSION TITLE
Prevent the file from being uploaded if the mimeType is invalid

### DISCUSSION THREAD

**Discussion by predaytor** [payloadcms/payload#937] [Feature Requests & Ideas]
Сurrently, regardless of the configuration, the data is loaded to the disk, even after displaying an invalid mimeType error.
```
upload: {
    staticURL: '/uploads',

    staticDir: path.resolve(__dirname, '../../uploads'),

    mimeTypes: ['image/*'],
}
```
![image](https://user-images.githubusercontent.com/32335730/184551305-1ad47b71-5226-4a03-bb81-9bbd08ea3254.png)

![image](https://user-images.githubusercontent.com/32335730/184551314-2b5bc76b-566d-4e05-ad45-39a525380287.png)

**Comment by denolfe** [comment]
Thank you for the report @predaytor . I was able to recreate this - moving to an issue. #940

**Comment by Tommy1223454** [comment]
This is exactly why I prefer treating uploads as:

receive -> inspect -> then store/promote

instead of writing first and only then deciding whether the file should have been accepted.

I built an OSS package around that pattern in Node.js:
https://github.com/pompelmi/pompelmi

The goal is to keep the first trust decision inside the application path before persistence, so invalid or suspicious uploads don’t automatically become part of local storage / object storage just because the request reached the server.

**Comment by zubricks** [comment]
Resolved with https://github.com/payloadcms/payload/pull/1212
